src/Drupal/DrupalExtension/Context/MessageContext.php

```diff
@@ -361,4 +361,55 @@ private function assertNot($message, $selectorId, $exceptionMsg)
             }
         }
     }
+
+  /**
+   * Check if the current page contains no messages of the specified selector.
+   *
+   * @Then /^I should see no (?P<selector>(?:success|error|warning)) message$/
+   */
+  public function assertSelectorNotVisible($selector):void {
+    if ($selector == 'success') {
+      $selector_id = 'success_message_selector';
+    }
+    else if (selector == 'error') {
+      $selector_id = 'error_message_selector';
+    }
+    else {
+      $selector_id = 'warning_message_selector';
+    }
+    $this->assertNotVisible($selector_id);
+  }
+
+  /**
+   * Checks if the current page contains neither error nor warning messages.
+   *
+   * @Then I should see neither error nor warning messages
+   */
+  public function assertErrorWarningNotVisible():void {
+    $this->assertNotVisible('error_message_selector');
+    $this->assertNotVisible('warning_message_selector');
+  }
+
+  /**
+   * Assert the selector is not visible.
+   *
+   * @param string $selector_id
+   *  The selector id to be checked.
+   *
+   * @throws \Behat\Mink\Exception\ExpectationException
+   *   Thrown when the unexpected message is present in the page.
+   */
+  private function assertNotVisible(string $selector_id):void {
+    $selector = $this->getDrupalSelector($selector_id);
+    $exception_msg = "The page '%s' contains the $selector_id '%s': %s";
+    $selector_objects = $this->getSession()->getPage()->findAll("css", $selector);
+    $visible_messages = [];
+    if (!empty($selector_objects)) {
+      foreach ($selector_objects as $selector_object) {
+        $visible_messages += $selector_object->getText();
+      }
+      throw new ExpectationException(sprintf($exception_msg, $this->getSession()->getCurrentUrl(), $selector_id, $visible_messages), $this->getSession()->getDriver());
+    }
```

Comment:
@kumikoono This should be:

```suggestion
    if (empty($selector_objects)) {
      return;
    }

    $visible_messages = [];
    foreach ($selector_objects as $selector_object) {
      if (empty($selector_object)) {
        continue;
      }
      $visible_messages += $selector_object->getText();
    }
    throw new ExpectationException(sprintf($exception_msg, $this->getSession()->getCurrentUrl(), $selector_id, $visible_messages), $this->getSession()->getDriver());
```

Reply:
Done.